AK/Optional.h

```diff
@@ -307,7 +323,29 @@ requires(!IsLvalueReference<T>) class [[nodiscard]] Optional<T> : public Optiona
     }
 
 private:
-    alignas(T) u8 m_storage[sizeof(T)];
+    ALWAYS_INLINE constexpr void construct_null_if_necessairy()
+    {
+        // OPTIMIZATION: Only construct the `m_null` member when we are constant-evaluating.
+        // Otherwise, this generates an unnecessairy zero-fill.
+#if defined(AK_COMPILER_GCC) || defined(HAS_ADDRESS_SANITIZER)
+        // NOTE: GCCs -Wuninitialized warning ends up checking this as well.
+        constexpr bool should_construct = true;
+#else
+        constexpr bool should_construct = is_constant_evaluated();
+#endif
+        if (should_construct)
+            m_null = {};
+    }
```

Comment:
See above and CxBytes comments on the serenity PR

Reply:
I just re-confirmed in Compiler Explorer, initializing `m_null` does indeed zero fill in cases where it shouldn't have to at -O3 optimizations on both Clang and GCC. I fixed the `constexpr bool should_construct = is_constant_evaluated();` though, since that apparently always evaluates to true.